server/fleet/policies.go

```diff
@@ -1,14 +1,113 @@
 package fleet
 
+import (
+	"errors"
+)
+
+// PolicyPayload holds data for policy creation.
+type PolicyPayload struct {
+	// QueryID allows creating a policy from an existing query.
+	//
+	// Using QueryID is the old way of creating policies.
+	// Use Query, Name and Description instead.
+	QueryID uint
+	// Name is the name of the policy (ignored if QueryID != 0).
+	Name string
+	// Query is the policy query (ignored if QueryID != 0).
+	Query string
+	// Description is the policy description text (ignored if QueryID != 0).
+	Description string
+	// Resolution indicate the steps needed to solve a failing policy.
+	Resolution string
+}
+
+var (
+	errPolicyEmptyName     = errors.New("policy name cannot be empty")
+	errPolicyEmptyQuery    = errors.New("policy query cannot be empty")
+	errPolicyIDAndQuerySet = errors.New("both fields \"queryID\" and \"query\" cannot be set")
+	errPolicyInvalidQuery  = errors.New("invalid policy query")
+)
+
+// Verify verifies the policy payload is valid.
+func (p PolicyPayload) Verify() error {
+	if p.QueryID != 0 {
+		if p.Query != "" {
+			return errPolicyIDAndQuerySet
+		}
+	} else {
+		if err := verifyPolicyName(p.Name); err != nil {
+			return err
+		}
+		if err := verifyPolicyQuery(p.Query); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func verifyPolicyName(name string) error {
+	if name == "" {
+		return errPolicyEmptyName
+	}
+	return nil
+}
+
+func verifyPolicyQuery(query string) error {
+	if query == "" {
+		return errPolicyEmptyQuery
+	}
+	if validateSQLRegexp.MatchString(query) {
+		return errPolicyInvalidQuery
+	}
+	return nil
+}
+
+// ModifyPolicyPayload holds data for policy modification.
+type ModifyPolicyPayload struct {
+	// Name is the name of the policy.
+	Name *string `json:"name"`
+	// Query is the policy query.
+	Query *string `json:"query"`
+	// Description is the policy description text.
+	Description *string `json:"description"`
+	// Resolution indicate the steps needed to solve a failing policy.
+	Resolution *string `json:"resolution"`
```

Comment:
It would be awesome if we could generalize this logic of Entity struct that has some data, EntityPayload that has basically the same data but it's pointers. If pointer is nil, leave that part untouched.